pkg/specgen/generate/security.go

```diff
@@ -144,10 +144,14 @@ func securityConfigureGenerator(s *specgen.SpecGenerator, g *generate.Generator,
 		configSpec.Process.Capabilities.Inheritable = caplist
 		configSpec.Process.Capabilities.Ambient = caplist
 	} else {
-		configSpec.Process.Capabilities.Effective = []string{}
+		userCaps, err := capabilities.NormalizeCapabilities(s.CapAdd)
+		if err != nil {
+			return errors.Wrapf(err, "capabilities requested by user are not valid: %q", strings.Join(s.CapAdd, ","))
+		}
+		configSpec.Process.Capabilities.Effective = userCaps
 		configSpec.Process.Capabilities.Permitted = []string{}
 		configSpec.Process.Capabilities.Inheritable = []string{}
```

Comment:
Should we be setting this too? It seems strange that they can't be inherited if we're giving them to the first process (esp. considering we set all 4 about for root)

Reply:
Yes I am not sure if we should set all 4 or just effective, or any combination.